cpp/dynamic_community_detection_module/algorithm/dynamic_community_detection.cpp

```diff
@@ -0,0 +1,371 @@
+#include <cmath>
+#include <mg_graph.hpp>
+
+#include "dynamic_community_detection.hpp"
+
+namespace LabelRankT {
+/* #region helper_methods */
+///@brief Checks if vector is subset of another vector.
+///
+///@tparam T -- set element data type
+///
+///@param v1 -- subset candidate
+///@param v2 -- vector against which the check is performed
+///
+///@return -- whether v1 is subset of v2
+template <class T>
+bool IsSubset(const std::vector<T>& v1, const std::vector<T>& v2) {
+  for (const auto& element : v1) {
+    if (std::find(v2.begin(), v2.end(), element) == v2.end()) return false;
+  }
+
+  return true;
+}
+
+std::vector<std::uint64_t> LabelRankT::NodesMemgraphIDs() {
+  std::vector<std::uint64_t> nodes;
+
+  for (const auto node : graph->Nodes()) {
+    nodes.push_back(graph->GetMemgraphNodeId(node.id));
+  }
+
+  return nodes;
+}
+
+std::unordered_set<std::uint64_t> LabelRankT::InNeighborsMemgraphIDs(
+    const std::uint64_t node_id) {
+  std::unordered_set<std::uint64_t> neighbors;
+
+  const auto neighbor_set =
+      this->is_directed ? graph->InNeighbours(graph->GetInnerNodeId(node_id))
+                        : graph->Neighbours(graph->GetInnerNodeId(node_id));
+  for (const auto node_i : neighbor_set) {
+    neighbors.insert(graph->GetMemgraphNodeId(node_i.node_id));
+  }
+
+  return neighbors;
+}
+
+void LabelRankT::SetStructures(const std::uint64_t node_id) {
+  times_updated[node_id] = 0;
+  std::unordered_map<std::uint64_t, double> node_label_Ps;
+
+  double sum_w_i = w_selfloop;
+  for (const auto node_j_id : InNeighborsMemgraphIDs(node_id)) {
+    sum_w_i += GetTotalWeightBetween(node_j_id, node_id);
```

Comment:
maybe `std::accumulate` ?

Reply:
Unfortunately, it would require an exceedingly complex lambda function.